.github/workflows/build.yaml

```diff
@@ -0,0 +1,61 @@
+name: Build
+on:
+    push:
+    pull_request:
+    workflow_dispatch:
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        type: [tar.xz, qcow2]
+        arch: [amd64, arm64]
+    env:
+      GITHUB_ACTION: true
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          # TODO: this should not be hardcoded
+          go-version: '1.24.2'
+
+      - name: Set up QEMU for arm64 builds
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver-opts: network=host
+
+      - name: Set up Docker layer cache
+        uses: actions/cache@v4
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
```

Comment:
Including `github.sha` means we always write to a new cache (but may restore the previous one as needed)? I guess that can make sense…